fsspec/implementations/local.py

```diff
@@ -206,24 +211,32 @@ def modified(self, path):
 
     @classmethod
     def _parent(cls, path):
-        path = cls._strip_protocol(path).rstrip("/")
-        if "/" in path:
-            return path.rsplit("/", 1)[0]
+        path = cls._strip_protocol(path, remove_trailing_slash=True)
+        if os.sep == "/":
+            # posix native
+            return path.rsplit("/", 1)[0] or "/"
         else:
-            return cls.root_marker
+            # NT
+            path_ = path.rsplit("/", 1)[0]
+            if len(path_) <= 3:
+                if path_[1:2] == ":":
+                    # nt root (something like c:/)
+                    return path_[0] + ":/"
+            # More cases may be required here
+            return path_
 
     @classmethod
-    def _strip_protocol(cls, path):
+    def _strip_protocol(cls, path, remove_trailing_slash=False):
         path = stringify_path(path)
-        if path.startswith("file://"):
-            path = path[7:]
-        elif path.startswith("file:"):
-            path = path[5:]
-        elif path.startswith("local://"):
-            path = path[8:]
+        if path.startswith("file:"):
```

Comment:
Should iterate over self.protocol (which can be a single string or a tuple, `("file", "local")` being the default for LocalFileSystem) to account for aliasing or subclasses.

Reply:
There is an *if* case for both `file` and `local`. In both cases the protocol will be stripped regardless of what `self.protocol` is set.

Reply:
Right, but I am thinking that there might be a subclass of LocalFileSystem that has a different protocol(s). Maybe that's not important - it would simply need to override this method

Reply:
I image reading a property and checking types will slow things down.